### PR TITLE
Theme sandbox as its own plugin

### LINKED ISSUE #9730: Split theme_sandbox into own plugin
Currently the theme_sandbox is part of the developers plugin. I think there are several reasons it should be stand alone.

The developers plugin is mainly used when developing, but there are cases where you wish to show off features / styling of your site without being in development mode (#7036). If we make it publically available (with a setting) we need to fix some issues like #9045.

I also would like to see the sandbox just being drawn in a normal page shell. That way it represents better the way the end result will look.

I also thought someone talked about reorganizing the contents of the sandbox, but i could not find the ticket for it

### PR DESCRIPTION
**Pull request by jdalsem**
feat(developers): theme_sandbox is now a standalone plugin

fixes #9730

chore(theme_sandbox): removed unsupported hover icons

---
- Theme sandbox now uses default layout and pageshell... on the layout page in the sandbox you can switch in which layout you wish to view all the sandbox pages (one_sidebar, one_column, two_sidebar)
- All sandbox pages are site menu items

### CODE REVIEW

**Comment by hypeJunction** [comment]
You should have merge 2.x into master before doing this

File: mod/theme_sandbox/views/default/theme_sandbox/layouts.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+echo 'Select the layout you wish to use looking at the sandbox pages:';
+echo '<ul><li>';
+echo elgg_view('output/url', [
+	'href' => elgg_http_add_url_query_elements(current_page_url(), [
+		'layout' => 'one_column',
+	]),
+	'text' => 'One Column',
+]);
+echo '</li><li>';
```
Comment (hypeJunction) [inline mod/theme_sandbox/views/default/theme_sandbox/layouts.php]: eek

File: mod/theme_sandbox/start.php [hunk 2]
```diff
@@ -0,0 +1,120 @@
+<?php
+/**
+ * Theme Sandbox
+ */
+
+elgg_register_event_handler('init', 'system', 'theme_sandbox_init');
+
+function theme_sandbox_init() {
```
Comment (hypeJunction) [inline mod/theme_sandbox/start.php]: We should either lock this down for admins or add a plugin setting.

File: mod/theme_sandbox/start.php [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+	elgg_register_plugin_hook_handler('register', 'menu:site', '_theme_sandbox_site_menu', 9999);
+	elgg_register_plugin_hook_handler('register', 'menu:page', '_theme_sandbox_page_menu');
+	
+	elgg_register_page_handler('theme_sandbox', 'theme_sandbox_controller');
```
Comment (hypeJunction) [inline mod/theme_sandbox/start.php]: It wishes it was a controller

File: mod/theme_sandbox/start.php [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+	
+	elgg_register_external_view('theme_sandbox/ajax'); // for lightbox in sandbox
+	
+	$sandbox_css = elgg_get_simplecache_url('theme_sandbox.css');
+	elgg_register_css('dev.theme_sandbox', $sandbox_css);
```
Comment (hypeJunction) [inline mod/theme_sandbox/start.php]: Why prefix?

**Comment by hypeJunction** [comment]
This is also going to break some pending PRs.

File: mod/theme_sandbox/start.php (hunk 4 above)
Comment (jdalsem) [inline mod/theme_sandbox/start.php]: kept it as it was

**Comment by jdalsem** [comment]
> You should have merge 2.x into master before doing this
> This is also going to break some pending PRs.

will rebase when 2.x is merged and pending blocking PRs are in

**Comment by mrclay** [comment]
If you're still into this, I think it'd be easier to re-create it than to resolve all the rebase conflicts. We can merge it quickly next time.

**Pull request closed without merging**